### PR TITLE
Added function that calls to CSL api 

### PR DESCRIPTION
**Pull request by microbe73**
Added the function which makes a call to the CSL api and returns the response from the api as a dictionary

### CODE REVIEW

File: amanuensis/config-default.yaml [hunk 1]
```diff
@@ -230,4 +230,8 @@ AWS_SES:
 HUBSPOT:
   API_KEY: "DEV_KEY"
   
-DB_MIGRATION_POSTGRES_LOCK_KEY: 100
\ No newline at end of file
+DB_MIGRATION_POSTGRES_LOCK_KEY: 100
+# //////////////////////////////////////////////////////////////////////////////////////
+# INSTITUTION: Used when calling the screening list API
+# //////////////////////////////////////////////////////////////////////////////////////
+INSTITUTION_NAME: ""
```
Comment (grugna) [inline amanuensis/config-default.yaml:237]: This should be a parameter in the function and not a config variable

File: amanuensis/resources/institution/__init__.py [hunk 2]
```diff
@@ -3,4 +3,40 @@
...
+    try:
+        url = f"https://data.trade.gov/consolidated_screening_list/v1/search?name={name}"
+
+        hdr ={
+        # Request headers
+        'Cache-Control': 'no-cache',
+        'subscription-key': 'a7c2cd5a313d430c9df8bc7918f2e14b',
+        }
+
+        req = urllib.request.Request(url, headers=hdr)
```
Comment (grugna) [inline amanuensis/resources/institution/__init__.py:28]: For consistency please use the same requests module we are using elsewhere `import requests`

File: amanuensis/resources/institution/__init__.py [hunk 3]
```diff
@@ -3,4 +3,32 @@
...
+)
+import requests, json
+def api_request(name):
+    """
+    Makes a call to the Consolidated Screening List api of developer.trade.gov. Information returned in the dictionary
+    can be accessed by info_dict["results"], which is a list of dictionaries containing information about the legality
+    of interacting with the returned companies.
+    """
+    try:
+        url = f"https://data.trade.gov/consolidated_screening_list/v1/search?name={name}"
```
Comment (grugna) [inline amanuensis/resources/institution/__init__.py:15]: Can you make this a variable in the config-default.yaml file?

File: amanuensis/resources/institution/__init__.py [hunk 4]
```diff
@@ -3,4 +3,32 @@
...
+        hdr ={
+        # Request headers
+        'Cache-Control': 'no-cache',
+        'subscription-key': 'a7c2cd5a313d430c9df8bc7918f2e14b',
```
Comment (grugna) [inline amanuensis/resources/institution/__init__.py:19]: is this subscription key personal or public?

move this to the config file, if private do not commit it and share it via https://onetimesecret.com

Comment (microbe73) [reply amanuensis/resources/institution/__init__.py]: It is a personal subscription key.
The website generated this link which has the API key: https://onetimesecret.com/secret/4kdalq354mhqouo7ch9apzsg5kkjwp
It is now no longer in the code and has to be manually entered into the config file, is this correct?

File: amanuensis/blueprints/admin.py [hunk 5]
```diff
@@ -172,6 +173,31 @@ def get_search_by_user_id():
 
     return jsonify({"filter_sets": filter_sets})
 
+@blueprint.route("/screen-institution", methods=["GET"])
+@check_arborist_auth(resource="/services/amanuensis", method = "*")
+def screen_institution():
+    name = request.args.get('name', default = "")
```
Comment (grugna) [inline amanuensis/blueprints/admin.py:179]: une None instead of ""

File: amanuensis/blueprints/admin.py [hunk 6]
```diff
@@ -172,6 +173,31 @@ def get_search_by_user_id():
...
+    if(name == ""):
```
Comment (grugna) [inline amanuensis/blueprints/admin.py:180]: Same as above None instead of ""

File: amanuensis/blueprints/admin.py [hunk 7]
```diff
@@ -172,6 +173,31 @@ def get_search_by_user_id():
...
+        raise UserError("Name of institution is needed in the name argument in the url")
+    res = api_request(name)
```
Comment (grugna) [inline amanuensis/blueprints/admin.py:182]: Rename this method from `api_request` to `get_background`

File: amanuensis/resources/institution/__init__.py [hunk 8]
```diff
@@ -3,4 +3,34 @@
 	get_all_institutions,
     update_institution,
     add_institution,
-)
\ No newline at end of file
+)
+import requests, json
+from amanuensis.config import config
+def api_request(name):
```
Comment (grugna) [inline amanuensis/resources/institution/__init__.py:9]: Rename this method from `api_request` to `get_background`

File: amanuensis/blueprints/admin.py [hunk 9]
```diff
@@ -172,6 +173,31 @@ def get_search_by_user_id():
...
+    if(total <= 0):
+        raise APIError("Institution not found in the API, double-check spelling")
+    if(total >= 10):
+        print("The API only returns 10 results at a time, but more results match the search. If searching for one particular institution you may need to be more specific about the name")
+    try:
+        first_result_dict = res["results"][0]
+        first_id = first_result_dict["id"]
+        first_name = first_result_dict["name"]
+    except:
+        raise APIError("Possible change to or error with CSL api, unable to access required fields. See https://developer.trade.gov/api-details#api=consolidated-screening-list")
```
Comment (grugna) [inline amanuensis/blueprints/admin.py:197]: What is this for?

File: amanuensis/resources/institution/__init__.py [hunk 10]
```diff
@@ -3,4 +3,34 @@
 	get_all_institutions,
     update_institution,
     add_institution,
-)
\ No newline at end of file
+)
+import requests, json
+from amanuensis.config import config
```
Comment (grugna) [inline amanuensis/resources/institution/__init__.py:8]: move the imports to the top of the file

File: amanuensis/blueprints/admin.py (hunk 9 above)
Comment (microbe73) [reply amanuensis/blueprints/admin.py:197]: These were a couple of checks to make sure the api hasn't changed the format or something, is that necessary or can I get rid of it?

**Comment by paulmurdoch19** [comment]
@microbe73 @grugna when starting amanuensis an import error is thrown so that piece needs to be added

**Comment by paulmurdoch19** [comment]
@grugna it also appears that Institution is not in userportaldatamodel.

**Comment by grugna** [comment]
@paulmurdoch19 good catch. It doesn't have a table by itself but lives in the Project table - https://github.com/chicagopcdc/userPortalDataModel/blob/845f09d28486471cd95b81f3b1c2151d27fd1a3a/userportaldatamodel/project.py#L40C1-L41C1

**Comment by grugna** [comment]
That userportaldatamodel_institution file was used prior to deciding to get the info about the institution from Hubspot, I guess we forgot to remove it.

**Pull request merged**